lib/puppet/provider/service/smf.rb

```diff
@@ -107,6 +107,8 @@ def start
   end
 
   def stop
+    # Don't try to stop non-existing services (PUP-8167)
+    return if self.status == :absent
```

Comment:
It looks like status returns `:absent` whenever the status command returns non-zero. https://github.com/ScottGarman/puppet/blob/1efbb277937d88c26948abc562f14e81ce315706/lib/puppet/provider/service/smf.rb#L141. Should we only return `:absent` for a specific exit code?

Reply:
@joshcooper I'll look into this.

Reply:
@joshcooper Technically the svdadm return code (1) in this case just means a fatal error occurred, check the error message output. There isn't a specific return code for "this service doesn't exist". I don't know that making :absent specific to exit code 1 really does what we want it to, and IMO this is a separate issue that probably needs its own ticket and some further reasoning.

https://www.unix.com/man-page/OpenSolaris/1m/svcadm/

Reply:
Looks good as is, thanks for checking.